dapp_user/infrastructure/repositories/user_repository.py

```diff
@@ -46,3 +46,15 @@ def get_user_preferences(self, user_row_id):
     def delete_user(self, username):
         query = "DELETE FROM user WHERE username = %s "
         self._repo.execute(query, [username])
+
+    def register_user_data(self, user):
+        """ register user data """
+        if not user.email_verified:
+            raise Exception("Email verification is pending.")
```

Comment:
We need to get the user even there is no email verification. If the hook is signup it might not fire in case if email verification.

Reply:
the hook we are using is a post-confirmation trigger, which mean email is verified.